lib/Unexpected.js

```diff
@@ -484,14 +482,30 @@ Unexpected.prototype.fail = function (arg) {
                 output.error('Explicit failure');
             }
         };
+        var expect = this.expect;
         Object.keys(arg).forEach(function (key) {
             var value = arg[key];
             if (key === 'diff') {
-                error.createDiff = value;
+                if (typeof value === 'function' && this.parent) {
```

Comment:
in assertions.js we have some cases where the `diff` property ends up having a value of `false`, eg. https://github.com/unexpectedjs/unexpected/blob/b2e122f0196d1a9a50e28d46ddef7f75d6a9abeb/lib/assertions.js#L287